Automattic/jetpack
# RNA Connection: Add ConnectScreenRequiredPlan Component

Adds the `ConnectScreenRequiredPlan` in `jetpack-connection` package.

#### Changes proposed in this Pull Request:

 **Connection js package:**
* Adds the `ConnectScreenRequiredPlan` along with its visual counterpart, stories and tests.
* Adds the `ConnectScreenLayout` helper layout component that contains the common logic between the two different Connect Screens
* Updates the `ConnectScreen`  adding its visual counterpart and making use of `ConnectScreenLayout`

 **Components js package:**
* Updates the `PricingCard` component to accept children, makes the CTA optional and fixes the prices PropTypes

 **Storybook js package:**
* Adds the connection-package to Storybook
* Adds a public folder for storing static assets, like images for Storybook stories.

**Connection-ui package**
* Replaces the `ConnectScreen` with `ConnectScreenRequiredPlan` for testing purposes.

#### Jetpack product discussion
1200295095943283-as-1201123230360951/f

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions (Design) - Phase 1:
**Test Storybook**

* Go to the Story book folder `cd projects/js-packages/storybook`
* `pnpm install`
* pnpm run storybook:dev
* Your browser should open with Storybook
* Go to Connection/Connect Screen with Required Plan and play around with it.

Design considerations (cc @keoshi )
- Checkout the **Errored** story in Storybook. The error displayed under the `Log in` link is not ideal. We are considering using notices in the future, however pls let me know if the current UI is not acceptable.
-  Make sure to test for various viewports. The design was only for desktops so I improvised a bit :)

#### Testing instructions (Crew) - Phase 2:
* Go to Tools -> Connection Manager, disconnect if needed, then connect again by clicking "Get Jetpack"
* Disconnect and connect again but this time by clicking "Log in"
* Disconnect, change `autoTrigger` to `true` and confirm you are automatically connected.
* Disconnect if needed, Go to Jetpack dashboard and confirm the Connect Screen hasn't changed and you can successfully connect your account. Make sure to test with a Secondary user as well.
* Go to Jetpack->Backup and repeat the above confirming nothing has changed there as well.

#### Hacks (to be addressed in follow-up PRs)
- Accept button props in `ConnectButton`. Right now I used a hacky css solution to make it show as a link.
- Move the error display logic from `ActionButton` to parent containers (also discussed [here](https://github.com/Automattic/jetpack/pull/21473))

#### Design logic
Before introducing the `ConnectScreenRequiredPlanVisual` I created the Story taking advantage of `msw` ([Mock Service Worker](https://storybook.js.org/addons/msw-storybook-addon)). MSW captures network requests and returns a mocked response.
However this didn't feel "clean" enough so I decided to introduce the `ConnectScreenRequiredPlanVisual` which is the visual counterpart of `ConnectScreenRequiredPlan`.
You might notice the latter takes advantage of the `renderConnectBtn` **render** prop, in order for its consumers to define the connect button component and how we want it rendered.
Storybook will pass an `ActionButton` component while `ConnectScreenRequiredPlan` a `ConnectButton` component.

~~_I've also considered initially a composite component that could take advantage of the existing `ConnectScreen` component but after breaking it down they are so different that it's not worth it._~~
I'm now using the `ConnectScreenLayout` helper layout component that contains the common logic between the two different Connect Screens
